Error compiling CRAS

```
name: artful
encrypted: no
Entering /mnt/stateful_partition/crouton/chroots/artful...
crouton: version 1-20170814083539~master:980f090f
release: artful
architecture: armhf
xmethod: xorg
targets: lxde,xorg,xiwi,core
host: version 9592.71.0 (Official Build) beta-channel veyron_speedy 
kernel: Linux localhost 3.14.0 #1 SMP PREEMPT Wed Jul 26 23:30:44 PDT 2017 armv7l armv7l armv7l GNU/Linux
freon: yes
Unmounting /mnt/stateful_partition/crouton/chroots/artful...
```

Updating crouton fails with this error:

```
Installing target audio...
Fetching CRAS (branch 6e9aa2b7a694ee79518f05d544999468c4fd7055)...
--2017-08-14 09:00:19--  https://chromium.googlesource.com/chromiumos/third_party/adhd/+archive/6e9aa2b7a694ee79518f05d544999468c4fd7055.tar.gz
Resolving chromium.googlesource.com (chromium.googlesource.com)... 108.177.97.82, 2404:6800:4008:c00::52
Connecting to chromium.googlesource.com (chromium.googlesource.com)|108.177.97.82|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/x-gzip]
Saving to: '/tmp/crouton-cras.pMpnIM/adhd.tar.gz'

     0K .......... .......... .......... .......... ..........  245K
    50K .......... .......... .......... .......... ..........  243K
   100K .......... .......... .......... .......... ..........  924K
   150K .......... .......... .......... .......... ..........  487K
   200K .......... .......... .......... .......... ..........  414K
   250K .......... .......... .......... .......... ..........  586K
   300K .......... .......... .......... .......... ..........  772K
   350K .......... .......... .......... .......... .......... 1.06M
   400K .......... .......... .......... .......... .......... 1.14M
   450K .......... .......... .......... .......... .......... 1.01M
   500K .......... .......... .......... .......... ..........  972K
   550K .......... .......... .......... .......... .......... 1.71M
   600K .......... ..........                                   696K=1.1s

2017-08-14 09:00:22 (573 KB/s) - '/tmp/crouton-cras.pMpnIM/adhd.tar.gz' saved [634976]

Reading package lists... Done
Building dependency tree       
Reading state information... Done
alsa-utils is already the newest version (1.1.3-1ubuntu1).
libasound2 is already the newest version (1.1.3-5).
libspeexdsp1 is already the newest version (1.2~rc1.2-1ubuntu1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
gcc is already the newest version (4:7.1.0-1ubuntu3).
libc6-dev is already the newest version (2.24-12ubuntu1).
libspeexdsp-dev is already the newest version (1.2~rc1.2-1ubuntu1).
pkg-config is already the newest version (0.29.1-0ubuntu2).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libasound2-dev is already the newest version (1.1.3-5).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
patch is already the newest version (2.7.5-1build1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Compiling CRAS (native)...
common/edid_utils.c: In function 'show_edid':
common/edid_utils.c:687:29: error: '*' in boolean context, suggest '&&' instead [-Werror=int-in-bool-context]
   (edid_data[EDID_VIDEO_IN] * 0x10) ? "Blank to black" : "",
   ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
common/edid_utils.c:688:29: error: '*' in boolean context, suggest '&&' instead [-Werror=int-in-bool-context]
   (edid_data[EDID_VIDEO_IN] * 0x08) ? "Separate" : "",
   ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
common/edid_utils.c:689:29: error: '*' in boolean context, suggest '&&' instead [-Werror=int-in-bool-context]
   (edid_data[EDID_VIDEO_IN] * 0x04) ? "Composite" : "",
   ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
common/edid_utils.c:690:29: error: '*' in boolean context, suggest '&&' instead [-Werror=int-in-bool-context]
   (edid_data[EDID_VIDEO_IN] * 0x02) ? "On-green" : "",
   ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
common/edid_utils.c:691:29: error: '*' in boolean context, suggest '&&' instead [-Werror=int-in-bool-context]
   (edid_data[EDID_VIDEO_IN] * 0x01) ? "Serration V" : "",
   ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
cc1: all warnings being treated as errors
Failed to complete chroot setup.
Unmounting /mnt/stateful_partition/crouton/chroots/artful...
Sending SIGTERM to processes under /mnt/stateful_partition/crouton/chroots/artful...
```

--- a/targets/audio
+++ b/targets/audio
@@ -172,8 +172,10 @@ END
         {
             convert_automake
 
+            # Note: -Wno-int-in-bool-context is necessary for building CRAS on
+            # older ChromeOS versions (issue #3352).
             echo '
-                CFLAGS="$CFLAGS -DCRAS_SOCKET_FILE_DIR=\"/var/run/cras\""
+                CFLAGS="$CFLAGS -Wno-int-in-bool-context -DCRAS_SOCKET_FILE_DIR=\"/var/run/cras\""
 
                 buildlib libcras
 

